google/cloud/pubsub/options.h

```diff
@@ -220,12 +220,38 @@ struct CompressionAlgorithmOption {
   using Type = int;
 };
 
+/**
+ * The maximum number of Open Telemetry span links.
+ *
+ * This option controls the number of links contained in a single publish span.
+ *
+ * @note Application developers should keep in mind that Cloud Trace
+ *    sets [limits][cloud-trace-quota-link] on the number of links per span
+ * (128). Additionally, Open Telemetry sets the default [limit][otel-quota-link]
+ * to 128.
+ *
+ * @ingroup pubsub-options
+ *
+ * @par Environment variable
+ * This option is controlled by the `OTEL_SPAN_LINK_COUNT_LIMIT`
```

Comment:
Do we want to re-use the Open Telemetry environment variable, or define one of our own?

Reply:
We are reusing the otel variable. I think that makes the most sense.